### ISSUE TITLE
TraitsUI minor version release - 7.1.0

### ISSUE THREAD

**Issue by rahulporuri** [enthought/traitsui#1307]
Checklist adapted from earlier issue for 7.0.1 release - https://github.com/enthought/traitsui/issues/933
Target date: Release Candidate by 09 Oct 2020, Release by 16 Oct 2020
This is a minor release. See https://github.com/enthought/pyface/issues/714 for Pyface release 7.1.0

Release candidate
-------------------
- [x] Verify that all relevant PRs have been merged to master.
- [x] Create a PR against master to bump version to 7.2, merge that PR (#1358)
- [x] From the commit just before bumping the version, create a new branch `maint/7.1`
- [x] Make sure `maint/7.1` is included (somehow) in the following block in appveyor.yml (#1361, see #1362 as well)
```
branches:
  only:
    - ...
```
- [x] Update changelog and open PR targeting a new `maint/7.1` branch (#1356)
- [x] Update `ci-src-requirements.txt` to target Pyface `maint/7.1` branch (#1360)
- [x] Check MANIFEST and requirements are still up to date.
- [x] Update version in setup.py for the prerelease, open a PR against maint/7.1 (#1381)
       - (FTR: This step was done for 7.1.0rc2 but not 7.1.0rc1 due to #1357)
       - Create a new branch from maint/7.1: `git checkout maint/7.1; git pull; git checkout -b call-it-anything-you-like`
       - Set PRERELEASE to "rc2" and IS_RELEASED to true, commit, e.g.git commit -m "Set IS_RELEASED to true for prerelease 7.1.0rc2"
       - Flip IS_RELEASED back to false, commit.
       - Open a PR against maint/7.1
       - Merge but DO NOT SQUASH
No squashing so that the tagged commit in the next step will be on the branch (Kit: It is not a big deal even if it was squashed. But some prefer having the commit on the branch.) Alternatively, flip the IS_RELEASED flag in a separate PR.
- [x] Tag (annotated!) the release candidate on the commit where IS_RELEASED is set to true, e.g. git tag -a -m "Release candidate 7.1.0rc2" 7.1.0rc2 <commit-hash>
       - (FTR: the tag for 7.1.0rc1 was different due to #1357)
- [x] Push the tag to GitHub
- [x] Upload to PyPI
       - (FTR: This step was done for 7.1.0rc2 but not 7.1.0rc1 due to #1357)
       - `git checkout <tag>`, `git clean -ffxd`, `python setup.py sdist`, `twine check dist/<...>.tar.gz`, `twine upload dist/<...>.tar.gz`
- [x] Announcement for the release candidate

Release blocker
----------------
- [x] #1355 (#1359)
- [x] #1378

Pre-release
---
- [x] Backport PRs that have been merged to master to the maintenance branch. Use the "need backport ..." tag if there is one (but don't rely 100% on it) (#1385)
- [x] Verify that no other open issue needs to be addressed before the release.
- [x] Test against other ETS packages and other traitsui-using projects
- [x] Check MANIFEST, requirements, changelog are still up to date.
- [x] Test building the documentation

Release
-------
- [x] Create branch release/7.1.0 from maint/7.1 branch. (#1388)
        - Set release to 7.1.0, and set IS_RELEASED is true; commit
        - Open a PR against maint/7.1 with this being the last commit so that CI is built on the release commit
        - Bump the version to 7.1.1 and set IS_RELEASED to false; commit, don't push yet.
        - Once CI is done building the last commit, push the commit
        - Merge but do not squash so that the release commit is on the branch (Kit: Again, not a big deal even if it was squashed.)
- [x] Restore the release/7.1.0 branch (Kit: not sure why we are keeping them, but we are it seems)
- [x] From the commit at which IS_RELEASED is true and version is 7.1.0, tag (annotated!) `git tag -a -m "Release 7.1.0" 7.1.0`
- [x] Push the tag `git push origin 7.1.0`
- [x] Make PR targeting `gh-pages` branch: Generate documentation and copy the content to the branch. Verify that the resulting index.html looks good. (#1389)
- [x] Install from source distribution and run tests again (Kit: perhaps this should be moved up to go before merging the release PR?)
- [x] Upload to PyPI
- [x] Test the PyPI package

Post-release
-------------
- [x] Package update for `enthought/free` repository (for EDM)
- [x] Backport release note and change log to master, and possibly `maint/7.1` branch. (#1390)
- [x] Update GitHub Release pages https://github.com/enthought/traitsui/releases
- [x] Announcement (e.g. Google Group)

**Comment by aaronayres35** [comment]
Closing as I just made the announcement which wraps up the issue checklist.
